mistralrs-core/src/pipeline/hf.rs

```
@@ -64,17 +78,58 @@ pub(crate) fn api_get_file(
         info!("Loading `{file}` locally at `{}`", path.display());
         Ok(path)
     } else {
-        api.get(file).map_err(|e| match e {
-            HFHubApiError::RequestError(err) if matches!(*err, ureq::Error::Status(403, _)) => {
-                HFError::AuthorizationError {
-                    model_id: model_id.display().to_string(),
+        // **Check for SIGTERM before starting the request**
+        if SHOULD_TERMINATE.load(AtomicOrdering::SeqCst) {
+            eprintln!("Download aborted due to SIGTERM.");
+            process::exit(1);
```

This is a bit heavy-handed - we should probably just return an Err